src/pip/_internal/resolution/resolvelib/requirements.py

```diff
@@ -104,8 +116,10 @@ def name(self):
         # type: () -> str
         return self._candidate.name
 
-    def find_matches(self):
-        # type: () -> Sequence[Candidate]
+    def find_matches(self, constraint):
+        # type: (Sequence[SpecifierSet]) -> Sequence[Candidate]
+        assert len(constraint) == 0, \
+            "RequiresPythonRequirement cannot have constraints"
```

Comment:
Same here, the argument type seems wrong.

Reply:
The type hint does look wrong here. I may have messed up a merge. Not sure how mypy let me get away with that... I'll fix it.

Reply:
mypy is only as good as the annotations. And, honestly, not very good at incremental typing. :(

Reply:
Yeah, no worries. I just prefer to blame my tools than admit I made a mistake :-)